drgn_tools/iscsi.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2024, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+"""
+Helper for iscsi
+"""
+from typing import Iterator
+
+from drgn import container_of
+from drgn import Object
+from drgn import Program
+from drgn.helpers.linux.list import list_for_each_entry
+
+from drgn_tools.scsi import host_module_name
+from drgn_tools.table import print_table
+from drgn_tools.util import has_member
+
+
+def for_each_iscsi_device(prog: Program) -> Iterator[Object]:
+    """
+    Iterates through all iscsi devices and returns a
+    iterator.
+    :returns: a iterator of ``struct dev *``
+    """
+    class_in_private = prog.type("struct device_private").has_member(
+        "knode_class"
+    )
+
+    devices = Object(
+        prog,
+        "struct class",
+        address=prog["iscsi_host_class"].address_of_().value_(),
+    ).p.klist_devices.k_list.address_of_()
+
+    if class_in_private:
+        for device_private in list_for_each_entry(
+            "struct device_private", devices, "knode_class.n_node"
+        ):
+            yield device_private.device
```

Comment:
In the code path of `class_in_private` is not null, it never checks anything for iscsi, why will it be iscsi device?